modules/storages/app/common/storages/peripherals/storage_interaction/nextcloud/files_query.rb

```diff
@@ -58,6 +58,12 @@ def query(data)
 
     private
 
+    def requested_folder(folder)
+      return '' if folder.nil?
+
+      folder.gsub(' ', '%20')
```

Comment:
`GCI.escape` will not work here, because we need to keep the slashes (`/`). Therefor I only replaced spaces.

I tried writing a spec for it, but for WebMock `/Folder With Spaces` and `/Folder%20With%20Spaces` are the same, because the get parsed into a URL model of some sort. Comparison is then done on these model not on the strings. The test, if possible, would look rather complex. Any ideas, or do we leave it out?

Reply:
leave the test out, I'd say.

just to understand the issue: The rails code already decoded the `%20`, then it gets here as a string with spaces, and before sending it to NC, we need to encode it again? Was that the bug?

Reply:
Yes.